add macOS build for libretro core

rtretiakov opened the pull request.

byuu commented:
I merged this but with one change: on macOS, -fopenmp works for me. I posit you might have an older compiler toolchain? It's a *really* big performance penalty to not have OpenMP. HD mode 7 can hit 8x scale on an i9 with it, but only 3x scale without it. I hope we can find a solution for that.

Also, I hope it's okay but I'd rather directly copy-merge these patches. Please continue sending them as pull requests. But the last time I merged one of these, it broke my GitLab bsnes mirroring, saying things diverged from master. I'm new to using Git, sorry. But I'll tag you with credit in every release. Is that alright with you? If not, then I'll try to figure it out sooner.

rtretiakov commented:
Whatever works for you as you get used to git is fine with me. It can take some time to learn all of the tricks needed to handle complex situations.

I am completely up to date on my macOS installation, but I can't seem to use -fopenmp. I have not done anything fancy there, just installed Xcode and some stuff from homebrew that's unrelated to my work in emulation. I'd of course like to understand how our toolchains are different. Do you use LLVM from homebrew maybe? I've done a bit of research about this just now, and it looks like that is a method you can use to make this flag work, in addition to doing brew install libomp.

One thing that was missed, I updated the resources.hpp file to be an exact copy of what you have in target-bsnes rather than the xxd dumps I did. I see Boards.bml was updated recently so I just copied it directly to stay in sync. I can give you another pull request for that if you want, or you can just take it directly from my fork, which I just rebased against yours. Whatever works.
